pkg/collector/py/py3_checker.go

```diff
@@ -0,0 +1,68 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2018 Datadog, Inc.
+
+// +build cpython
+
+package py
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/DataDog/datadog-agent/pkg/util/executable"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+func init() {
+	here, _ := executable.Folder()
+	py3LinterPath = filepath.Join(here, py3LinterPath)
+	if _, err := os.Stat(py3LinterPath); err == nil {
+		log.Debugf("python3 linter found (%s): enabling linter", py3LinterPath)
+	} else {
+		log.Warnf("could not find python3 linter '%s': disabling linter", py3LinterPath)
+		py3LinterPath = ""
+	}
+}
+
+type warning struct {
+	Message string
+}
+
+// verifyPython3 checks that a check can run on python 3
+func validatePython3(moduleName string, modulePath string) ([]string, error) {
+	if py3LinterPath == "" {
+		return nil, nil
```

Comment:
I think we should return an error here, if some code called `validatePython3` and something went wrong at `init` time, the caller should check whether `res` is `nil` to see if the validation happened or not, which seems a bit weird.